paper-api/src/main/java/io/papermc/paper/raytracing/PositionedRayTraceConfigurationBuilder.java

```diff
@@ -0,0 +1,184 @@
+
+    /**
+     * Gets if the raytrace will ignore passable blocks when looking for block collisions.
+     *
+     * @return if the raytrace will ignore passable blocks
+     */
+    boolean ignorePassableBlocks();
+
+    /**
+     * Gets if the raytrace will ignore passable blocks when looking for block collisions.
+     *
+     * @param ignorePassableBlocks if the raytrace should ignore passable blocks
+     * @return a reference to this object
+     */
+    @NotNull
+    @Contract("_ -> this")
+    PositionedRayTraceConfigurationBuilder ignorePassableBlocks(boolean ignorePassableBlocks);
+
+    /**
+     * Gets the size of the raytrace when looking for entity collisions.
+     *
+     * @return the raytrace size
+     */
+    double raySize();
+
+    /**
+     * Sets the size of the raytrace when looking for entity collisions.
+     *
+     * @param raySize the new raytrace size
+     * @return a reference to this object
+     */
+    @NotNull
+    @Contract("_ -> this")
+    PositionedRayTraceConfigurationBuilder raySize(double raySize);
+
+    /**
+     * Gets the current entity filter when looking for entity collisions.
+     *
+     * @return predicate for entities the ray can potentially collide with, or null to consider all entities
+     */
+    @Nullable
+    Predicate<? super Entity> entityFilter();
+
+    /**
+     * Sets the current entity filter when looking for entity collisions.
+     *
+     * @param entityFilter predicate for entities the ray can potentially collide with, or null to consider all entities
+     * @return a reference to this object
+     */
+    @NotNull
+    @Contract("_ -> this")
+    PositionedRayTraceConfigurationBuilder entityFilter(@Nullable Predicate<? super Entity> entityFilter);
+
+    /**
+     * Gets the current block filter when looking for block collisions.
+     *
+     * @return predicate for blocks the ray can potentially collide with, or null to consider all blocks
+     */
+    @Nullable
+    Predicate<? super Block> blockFilter();
+
+    /**
+     * Sets the current block filter when looking for block collisions.
+     *
+     * @param blockFilter predicate for blocks the ray can potentially collide with, or null to consider all blocks
+     * @return a reference to this object
+     */
+    @NotNull
+    @Contract("_ -> this")
+    PositionedRayTraceConfigurationBuilder blockFilter(@Nullable Predicate<? super Block> blockFilter);
+
+    /**
+     * Gets the current set targets.
+     *
+     * @return the targets
+     */
+    @Nullable
+    List<Targets> targets();
+
+    /**
+     * Sets the targets for the rayTrace.
+     *
+     * @return a reference to this object
+     */
+    @NotNull
+    @Contract("_, _ -> this")
+    PositionedRayTraceConfigurationBuilder targets(@NotNull Targets first, @NotNull Targets... others);
+
+    /**
+     * List of Targets the builder can target.
+     */
+    enum Targets {
+        ENTITIES,
+        BLOCKS
+    }
```

Comment:
Lets put this in a separate enum inside the package. If we ever add the buildable/reusable builder that doesn't have location/direction, we will regret having this enum inside this type. Make it just like `RayTraceTarget` with `ENTITY` and `BLOCK` or something.